dapp/src/components/SetProject.astro

```diff
@@ -56,23 +57,60 @@ const TANSU_CONTRACT_ID = import.meta.env.PUBLIC_TANSU_CONTRACT_ID;
       "[data-set-project]",
     ) as HTMLButtonElement;
 
-    if (button) {
-      button.addEventListener("click", async () => {
-        try {
-          setProjectId(project_name.value);
-          const project = await getProject();
-          if (project && project.name && project.config && project.maintainers) {
-            setProject(project);
-            const { username, repoName } = getAuthorRepo(project.config.url);
-            if (username && repoName) {
-              setProjectRepoInfo(username, repoName);
-            }
-            navigate("/commit");
-          } else {
-            navigate("/register")
+    // Function to handle project setting
+    const handleSetProject = async () => {
+      try {
+        setProjectId(project_name.value.toLowerCase());
+        const project = await getProject();
+        if (project && project.name && project.config && project.maintainers) {
+          setProject(project);
+          const { username, repoName } = getAuthorRepo(project.config.url);
+          if (username && repoName) {
+            setProjectRepoInfo(username, repoName);
+          }
+          const tomlData = await fetchTOMLFromConfigUrl(project.config.url);
```

Comment:
Can you check the caching or default values? I tested with another project and I get data from tansu still.

<img width="766" alt="Screenshot 2024-09-12 at 01 01 18" src="https://github.com/user-attachments/assets/30211030-1492-4d2a-97e7-94a40886159c">

Reply:
this bug comes from you can't read config data from github
I set the config data url as 'main/tansu.toml'
in scipy repo, I can't find the config data toml file